### PR TITLE
PP-4374 Remove NotificationService and StatusMappers

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT

Part of refactoring notification handling (see https://github.com/alphagov/pay-connector/pull/787 for the original full attempt...). 
- Removed `NotificationService` : Payment provider specific notification services are in place now to handle notifications. `NotificationService` which was used to process notifications is no longer required.
- Removed `StatusMappers` : Status mapping is now incorporated within relevant NotificationServices except for Smartpay.  StatusMappers for Worldpay and EQPQ are not required.
-  `WorldpayNotificationServiceTest` : Updated/Added tests for all world refund notifications statuses

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gateway/smartpay/SmartpayPaymentProvider.java [hunk 1]
```diff
@@ -153,7 +151,7 @@ public boolean verifyNotification(Notification<Pair<String, Boolean>> notificati
 
     @Override
     public StatusMapper<Pair<String, Boolean>> getStatusMapper() {
-        return SmartpayStatusMapper.get();
+        return null;
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/gateway/smartpay/SmartpayPaymentProvider.java:154]: This method will be removed in next PR. Returning null as StatusMappers have been removed or not required here

File: src/main/java/uk/gov/pay/connector/gateway/epdq/EpdqPaymentProvider.java [hunk 2]
```diff
@@ -226,7 +225,7 @@ public ExternalChargeRefundAvailability getExternalChargeRefundAvailability(Char
 
     @Override
     public StatusMapper<String> getStatusMapper() {
-        return EpdqStatusMapper.get();
+        return null;
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/gateway/epdq/EpdqPaymentProvider.java:228]: This method will be removed in next PR. Returning null as StatusMappers have been removed or not required here

File: src/main/java/uk/gov/pay/connector/gateway/worldpay/WorldpayPaymentProvider.java [hunk 3]
```diff
@@ -168,7 +167,7 @@ public ExternalChargeRefundAvailability getExternalChargeRefundAvailability(Char
 
     @Override
     public StatusMapper<String> getStatusMapper() {
-        return WorldpayStatusMapper.get();
+        return null;
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/gateway/worldpay/WorldpayPaymentProvider.java:170]: This method will be removed in next PR. Returning null as StatusMappers have been removed or not required here

**Review by georgeracu — APPROVED** [review]
LGTM

**Pull request merged**